src/mantine-form/src/use-prop-ref/use-prop-ref.ts

```diff
@@ -0,0 +1,8 @@
+import React, { useRef } from 'react';
+
+export function usePropRef<T>(value: T): React.MutableRefObject<T> {
+  const ref = useRef(value);
+  ref.current = value;
```

Comment:
You should wrap this assignment in a useLayoutEffect with no dependency 🙂

While this works fine in React 17 and lower, it will create issues with React 18 where renders and hooks can be executed out of order and discarded due to internal optimizations in async mode